CONFIGURATION.md

```diff
@@ -0,0 +1,84 @@
+# Configuration examples
+On this page we will collect some configuration examples. This is meant to help you configure debug templates for your editor. The examples are provided in JSON format, but should be easily translatable to VSCode launch.json files, Emacs dap-mode templates, etc.
+
+
+**NOTE: Some editors like Emacs using lsp-mode will require you to set a `noDebug` argument to `false`/`nil` to start a debug process, if not you will simply run the code.**
+
+
+**NOTE 2: A general tip for working with the debug adapter is building your codebase before launching the debugger. A lot of people experiencing "class not found"-style issues, and they are in most cases caused by not building before debugging. A simple Maven or Gradle compile should suffice.**
+
+
+## launch
+### Regular main method
+Replace the `mainClassName` with your fully qualified class name where your main method resides (e.g, `com.example.MyApplicationKt`), and `projectRootPath` with the root of your project. If your main method resides in a file, the class name will be the package and name of the file (+ Kt at the end). 
+
+
+**Note:** If you use [the VSCode Kotlin extension](https://github.com/fwcd/vscode-kotlin) or [Emacs lsp-mode](https://emacs-lsp.github.io/lsp-mode/), you will have the option to use code lenses in your editor to run or debug main methods. These will fill out the details below automatically for you.
+
+
+```json
+{
+    "type": "kotlin",
+    "request": "launch",
+    "mainClass": mainClassName,
+    "projectRoot": projectRootPath
+}
+```
+
+
+### Test
+Debugging or running tests using the debug adapter might seem a bit daunting at first. What should the `mainClass` be? Fortunately it is quite simple if you use JUnit, as it provides a [Console Launcher](https://junit.org/junit5/docs/current/user-guide/#running-tests-console-launcher) that we can utilize.
+
+```json
+{
+    "type": "kotlin",
+    "request": "launch",
+    "mainClass": "org.junit.platform.console.ConsoleLauncher --scan-class-path",
+    "projectRoot": projectRootPath
+}
+```
+
+If you want to add logging to file, you can also add the following parameters:
```

Comment:
Perhaps this could use its own heading?

Because logging is not just test related right?
You are logging the messages send back and forth between the DAP-server and DAP-client.

From what I can think of its only really useful for debugging issues in communication between the two.

_Like when your dap-client requests stack-traces from threads that have not been stopped and the DAP-server returns a bunch of exceptions._ Not that this is a think I'm currently dealing with :angel:

Reply:
You are absolutely right! This is a brainfart from my end, probably due to me last using logging when doing tests. Fixed now 🙂

I notice that if you do println statements in your code, it logs to the same json log file 🙂 Only thing I have really used it for, even though I also get a status buffer with all of this info in Emacs even when no logging is enabled.